lib/Sema/CSSimplify.cpp

```diff
@@ -10987,6 +10987,11 @@ bool ConstraintSystem::resolveClosure(TypeVariableType *typeVar,
     if (contextualTy->isTypeVariableOrMember())
       return false;
 
+    // Cannot propagate pack expansion type from context,
+    // it has to be handled by type matching logic.
+    if (contextualTy->is<PackExpansionType>())
+      return false;
```

Comment:
If I understand everything correctly we could allow that by using `getPatternType` but I think in vast majority of cases (if not all) it would result in a type variable anyway which is not a suitable type too.

Reply:
We would need to wait until the pattern type is fully substituted and flattened. The pattern type in this case will be substituted with `PackType{String}`, and the pack expansion will flatten the pack type into just `String`

Reply:
Right, I figured that it better to let it go through the matching logic than try to do anything fancy here.